test/unit/fact_importer_test.rb

```diff
@@ -27,5 +32,30 @@ class CustomImporter < FactImporter; end
     test 'importers without authorized_smart_proxy_features return empty set of features' do
       assert_equal [], FactImporter.importer_for(:custom_importer).authorized_smart_proxy_features
     end
+
+    context 'importing facts' do
+      setup do
+        disable_orchestration
+        User.current = users :admin
+        @host = FactoryGirl.create(:host)
+      end
+
+      test 'facts of other type do not collide even if they inherit from FactName' do
+        assert_nothing_raised do
+          custom_import '_timestamp' => '234'
```

Comment:
Do we even need the helper methods since they are used just here? What about going just with 

```ruby
CustomImporter.new(@host, '_timestamp' => '234').import!
PuppetImporter.new(@host, '_timestamp' => '234').import!
```